Mathlib/Analysis/BoxIntegral/Basic.lean

```diff
@@ -669,13 +669,186 @@ end Integrable
 
 open MeasureTheory
 
+/-!
+### Oscillation
+-/
+
+section oscillation
+
+open EMetric
+
+/-- The oscillation of `f` at `x`. -/
+noncomputable def oscillation (f : ℝⁿ → E) (x : ℝⁿ) : ENNReal :=
+  ⨅ S ∈ (𝓝 x).map f, EMetric.diam S
+
+/-- The oscillation of `f` at `x` is 0 whenever `f` is continuous at `x`. -/
+theorem oscillation_zero_of_continuousAt {f : ℝⁿ → E} {x : ℝⁿ} (hf : ContinuousAt f x) :
+    oscillation f x = 0 := by
+  refine le_antisymm (ENNReal.le_of_forall_pos_le_add fun ε hε _ ↦ ?_) (zero_le _)
+  rw [zero_add]
+  have : EMetric.ball (f x) (ε / 2) ∈ (𝓝 x).map f :=
+    hf <| EMetric.ball_mem_nhds _ (by simp [ne_of_gt hε])
+  refine (biInf_le EMetric.diam this).trans (le_of_le_of_eq diam_ball ?_)
+  refine (ENNReal.mul_div_cancel' ?_ ?_) <;> norm_num
+
+/-- If `oscillation f x < ε` at every `x` in a compact set `K`, then there exists `δ > 0` such
+that the oscillation of `f` on `Metric.ball x δ` is less than `ε` for every `x` in `K`. -/
+theorem uniform_oscillation_of_compact {K : Set ℝⁿ} (comp : IsCompact K) (f : ℝⁿ → E)
+    (ε : ENNReal) (hK : ∀ x ∈ K, oscillation f x < ε) :
```

Comment:
```suggestion
theorem uniform_oscillation_of_compact {K : Set ℝⁿ} (comp : IsCompact K) {f : ℝⁿ → E}
    {: ENNReal} (hK : ∀ x ∈ K, oscillation f x < ε) :
```

`f` and `ε` can be implicit here

Reply:
@xroblot I think you still want the `ε` here, even if it's implicit.

Reply:
Yes, you're right